licensing/README.md

```diff
@@ -1,3 +1,66 @@
 # Licensing
 
-Placeholder for Licensing subproject
\ No newline at end of file
+This documents the Licensing subproject under SIG Release.
+The subproject is responsible for analyzing, reporting and remediating
+licensing concerns within the Kubernetes project orgs.
```

Comment:
Note to reviewer: This description comes from https://github.com/kubernetes/community/tree/master/sig-release#subprojects.